pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -0,0 +1,178 @@
+package image
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/diskfs/go-diskfs/filesystem"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+)
+
+const (
+	initrdimg = "initrd"
+	rootfsimg = "rootfs"
+	vmlinuz   = "vmlinuz"
+	// pxeAssetsPath is the path where pxe files are created.
+	pxeAssetsPath = "pxe"
+)
+
+// AgentPXEFiles is an asset that generates the bootable image used to install clusters.
+type AgentPXEFiles struct {
+	imageReader isoeditor.ImageReader
+	cpuArch     string
+	isoPath     string
+}
+
+var _ asset.WritableAsset = (*AgentPXEFiles)(nil)
+
+// Dependencies returns the assets on which the AgentPXEFiles asset depends.
+func (a *AgentPXEFiles) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&Ignition{},
+		&BaseIso{},
+	}
+}
+
+// Generate generates the image files for PXE asset.
+func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
+	ignition := &Ignition{}
+	dependencies.Get(ignition)
+
+	baseImage := &BaseIso{}
+	dependencies.Get(baseImage)
+
+	a.isoPath = baseImage.baseISOPath
```

Comment:
Could you use baseImage.File.Filename here instead of baseImage.baseISOPath?

Reply:
Yes, that seems to work. For some reason, I thought it won't give me the full path from the cache, which is not the case.